### PR TITLE
Fix issue deleting multiple blocks in document

### PR DESCRIPTION
**Pull request by Devorein**
### WHY

[Card](https://discord.com/channels/894960387743698944/1225077704865484810)

### CODE REVIEW

File: components/common/CharmEditor/components/rowActions/RowActionsMenu.tsx [hunk 1]
```diff
@@ -72,8 +72,16 @@ function Component({ menuState }: { menuState: PluginState }) {
     }
   }
 
-  const isHeadingNode =
-    menuState.rowPos !== undefined ? view.state.doc.resolve(menuState.rowPos)?.node()?.type.name === 'heading' : false;
+  let isHeadingNode = false;
+  try {
+    // When a bunch of block is selected and removed, the rowPos goes out of range and throws an error
+    isHeadingNode =
+      menuState.rowPos !== undefined
```
Comment (mattcasey) [inline components/common/CharmEditor/components/rowActions/RowActionsMenu.tsx:79]: i think we can check for this instead , with doc.content.size > menuState.rowPos right?

File: __e2e__/charmEditor/rangeDeleteBlocks.spec.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  await documentPage.goToPage({
+    domain: space.domain,
+    path: page.path
+  });
+
+  await documentPage.charmEditor.click();
+  await documentPage.page.keyboard.press(`${process.platform === 'darwin' ? 'Meta' : 'Control'}+A`);
+  await documentPage.page.keyboard.press('Backspace');
+
+  await documentPage.page.waitForTimeout(500);
```
[the remark this answers is not in the document]
Comment (Devorein) [reply __e2e__/charmEditor/rangeDeleteBlocks.spec.ts:36]: Let me check if it works without a timeout. I'm not sure what we can wait for.

[the remark this answers is not in the document]
Comment (Devorein) [reply __e2e__/charmEditor/rangeDeleteBlocks.spec.ts:36]: Yep we need a timeout. Its not really possible in this case other than waiting for timeout, as we can't keep calling the prisma using `waitForFunction` method

File: components/common/CharmEditor/components/rowActions/RowActionsMenu.tsx (hunk 1 above)
Comment (Devorein) [reply components/common/CharmEditor/components/rowActions/RowActionsMenu.tsx:79]: Good point. Updated.

File: __e2e__/charmEditor/rangeDeleteBlocks.spec.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Devorein) [reply __e2e__/charmEditor/rangeDeleteBlocks.spec.ts]: Fixed the issue by checking the websocket `framereceived` payload

**Pull request merged**